File: C++/Dwie wieże.cpp
#include <string>
#include <iostream>

int main() 
{
    std::string rook1, rook2;
    std::cin >> rook1 >> rook2;
    std::cout << (rook1[0] == rook2[0] || rook1[1] == rook2[1] ? "TAK" : "NIE");
    return 0;
}

File: Java/Dwie wieże.java
import java.util.Scanner;

public class Main 
{
    public static void main(String[] args) 
    {
        Scanner scanner = new Scanner(System.in);
        String rook1 = scanner.next();
        String rook2 = scanner.next();
        System.out.println((rook1.charAt(0) == rook2.charAt(0) || rook1.charAt(1) == rook2.charAt(1)) ? "TAK" : "NIE");
    }
}